### ISSUE TITLE
Problems about pytorch and drjit compatibility

### ISSUE THREAD

**Issue by wawawa8** [mitsuba-renderer/mitsuba3#914]
## Summary

I would like to use pytorch inside a custom python bsdf model (e.g. using an MLP which takes in the incident and outgoing directions and the surface interaction coordinate, and output the bsdf value), and use mitsuba renderer for rendering, and use pytorch for training the model. But I find a problem passing the gradient.

The whole model is something like: (pytorch MLP inside mitsuba renderer) inside pytorch training framework. Refering to the guide of 'mitsuba and pytorch compability', I add a decorator outside the mitsuba.render function, which changes the output from drjit to torch tensors. However, after adding the decorator, the gradient passing of the mlp inside the bsdf is cut off. I tried to add another decorator outside the mlp forward function, but it seems useless.

A small demo is shown below.

![image](https://github.com/mitsuba-renderer/mitsuba3/assets/49435282/a60d6219-9b18-433d-8116-4ab83292aba7)

## System configuration

System information:

  OS: Ubuntu 22.04
  CPU: ...
  GPU: Nvidia RTX 4090
  Python version: 3.10.9
  LLVM version: ...
  CUDA version: ...
  NVidia driver: ...

  Dr.Jit version: ...
  Mitsuba version: ...
     Compiled with: ...
     Variants compiled: ...

**Comment by wjakob** [comment]
Dear @wawawa8,

sadly this is not supported. You can find more details in similar GitHub tickets requesting the same feature that have been closed in the past.
